disnake/poll.py

```diff
@@ -0,0 +1,415 @@
+        The layout type of the poll. Defaults to :attr:`PollLayoutType.default`.
+
+    Attributes
+    ----------
+    channel: Optional[Union[:class:`TextChannel`, :class:`VoiceChannel`, :class:`StageChannel`, :class:`Thread`, :class:`DMChannel`, :class:`GroupChannel`, :class:`PartialMessageable`]]
+        The channel that the poll was sent from. This will be None only when the object
+        is builded manually. The library always provide this attribute.
+
+        .. note::
+
+            This attribute is available only if you fetched the Poll object from the API.
+
+    message: Optional[:class:`Message`]
+        The message which contain this poll. This will be None only when the object
+        is builded manually. The library always provide this attribute.
+
+        .. note::
+
+            This attribute is available only if you fetched the Poll object from the API.
+
+    question: Union[:class:`str`, :class:`PollMedia`]
+        The question of the poll.
+    duration: Optional[:class:`datetime.timedelta`]
+        The remaining duration for this poll. ``None`` if already expired.
+    allow_multiselect: :class:`bool`
+        Whether users are able to pick more than one answer.
+    layout_type: :class:`PollLayoutType`
+        The type of the layout of the poll.
+    is_finalized: :class:`bool`
+        Whether the votes have been precisely counted.
+    """
+
+    __slots__ = (
+        "_state",
+        "channel",
+        "message",
+        "question",
+        "_answers",
+        "duration",
+        "allow_multiselect",
+        "layout_type",
+        "is_finalized",
+    )
+
+    def __init__(
+        self,
+        question: Union[str, PollMedia],
+        *,
+        answers: List[Union[str, PollAnswer]],
+        duration: timedelta = timedelta(hours=24),
+        allow_multiselect: bool = False,
+        layout_type: PollLayoutType = PollLayoutType.default,
+    ) -> None:
+        self._state: Optional[ConnectionState] = None
+        self.channel: Optional[MessageableChannel] = None
+        self.message: Optional[Message] = None
+
+        if isinstance(question, str):
+            self.question = PollMedia(question)
+        elif isinstance(question, PollMedia):
+            self.question: PollMedia = question
+        else:
+            raise TypeError(
+                f"Expected 'str' or 'PollMedia' for 'question', got {question.__class__.__name__!r}."
+            )
+
+        self._answers: Dict[int, PollAnswer] = {}
+        for answer in answers:
+            if isinstance(answer, PollAnswer):
+                self._answers[len(self._answers) + 1] = answer
+            elif isinstance(answer, str):
+                self._answers[len(self._answers) + 1] = PollAnswer(PollMedia(answer))
+            else:
+                raise TypeError(
+                    f"Expected 'List[str]' or 'List[PollAnswer]' for 'answers', got List[{answer.__class__.__name__!r}]."
+                )
```

Comment:
Idk if these types checks are consistent with the library, but i feel that we should validate them, let me know 🙏🏻

Reply:
Seems fine, I don't really see a way to rewrite this any better :+1: